Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/DefaultChartDataCollection.java

```diff
@@ -359,194 +234,113 @@ public DefaultChartDataCollection(APMLog apmLog) {
-                int y = avgPTMap.get(key);
-                this.averageProcessingTimeSeriesData.put(key, y);
-            }
+            this.averageProcessingTimeSeriesData = new UnifiedMap<>(avgPTMap);
+
             /**
              * Make max processing time chart series
              */
-            this.maxProcessingTimeSeriesData = new UnifiedMap<>();
-            for(double key : maxPTMap.keySet()) {
-                int y = maxPTMap.get(key);
-                this.maxProcessingTimeSeriesData.put(key, y);
-            }
+            this.maxProcessingTimeSeriesData = new UnifiedMap<>(maxPTMap);
+
             /**
              * Make total waiting time chart series
              */
-            this.totalWaitingTimeSeriesData = new UnifiedMap<>();
-            for(double key : ttlWTMap.keySet()) {
-                int y = ttlWTMap.get(key);
-                this.totalWaitingTimeSeriesData.put(key, y);
-            }
+            this.totalWaitingTimeSeriesData = new UnifiedMap<>(ttlWTMap);
+
             /**
              * Make average waiting time chart series
              */
-            this.averageWaitingTimeSeriesData = new UnifiedMap<>();
-            for(double key : avgWTMap.keySet()) {
-                int y = avgWTMap.get(key);
-                this.averageWaitingTimeSeriesData.put(key, y);
-            }
+            this.averageWaitingTimeSeriesData = new UnifiedMap<>(avgWTMap);
+
             /**
              * Make max waiting time chart series
              */
-            this.maxWaitingTimeSeriesData = new UnifiedMap<>();
-            for(double key : maxWTMap.keySet()) {
-                int y = maxWTMap.get(key);
-                this.maxWaitingTimeSeriesData.put(key, y);
-            }
+
+            this.maxWaitingTimeSeriesData = new UnifiedMap<>(maxWTMap);
+
             /**
              * Make case utilization chart series
              */
-            this.caseUtilizationSeriesData = new UnifiedMap<>();
-            for(Double key : caseUtilMap.keySet()) {
-                int y = caseUtilMap.get(key);
-                this.caseUtilizationSeriesData.put(key, y);
+            this.caseUtilizationSeriesData = new UnifiedMap<>(caseUtilMap);
+        }
+    }
+
+
+
+    private void insertValuesToMaps(double[] valuesArray, DoubleArrayList[] keysArray,
+                                    List<TreeSortedMap<Double, Integer>> seriesMapsList) {
+        for (int i = 0; i < valuesArray.length; i++) {
+            insertValueToMap(valuesArray[i], keysArray[i], seriesMapsList.get(i));
+        }
+    }
+
+    private void insertValueToMap(double value, DoubleArrayList keys, TreeSortedMap<Double, Integer> seriesMap) {
+        for (int j = 0; j < keys.size(); j++) {
+            double key = keys.get(j);
+            if (value <= key) {
+                int y = seriesMap.get(key) + 1;
+                seriesMap.put(key, y);
+                break;
             }
         }
```

Comment:
I'm a little hazy on whether DoubleArrayList implements Iterable, but if it does you might be able to avoid the index j in this iteration:
```
for (double key: keys) {
    if (value <= key) {
       // etc
   }
}
```